### PR TITLE
api: update api methods/types/responses for go-threads

### LINKED ISSUE #180: Update to v0.1.3 API

### LINKED ISSUE #181: Add docs build; publish to gh-pages

### PR DESCRIPTION
**Pull request by andrewxhill**
fixes #180 
fixes #181

### CODE REVIEW

File: src/models.ts [hunk 1]
```diff
@@ -1,3 +1,29 @@
+import { Credentials } from '@textile/threads-client-grpc/api_pb'
+import { ThreadID } from '@textile/threads-core'
+/**
+ * Credentials for a Thread DB
+ */
+export class Creds {
```
Comment (andrewxhill) [inline src/models.ts]: A simple credentials object

File: src/models.ts [hunk 2]
```diff
@@ -1,3 +1,29 @@
...
+    if (this.signature) {
+      creds.setSignature(this.signature)
+    }
+    return creds
+  }
+
+  static defaultWithRandomID() {
+    const threadId = ThreadID.fromRandom()
+    const id = threadId.bytes()
+    return new Creds(id)
```
Comment (andrewxhill) [inline src/models.ts]: curious if you are thinking about anything nicer for credentials/threadid relationship/organization in JS. happy to change this to match whatever you do.

**Comment by carsonfarmer** [comment]
Assuming we are holding off on this one until we get some consensus on https://github.com/textileio/go-threads/pull/293?

File: package.json [hunk 3]
```diff
@@ -57,47 +57,47 @@
...
     "eslint-config-strict": "^14.0.1",
-    "eslint-plugin-import": "^2.18.2",
-    "eslint-plugin-node": "^11.0.0",
+    "eslint-plugin-import": "^2.20.2",
+    "eslint-plugin-node": "^11.1.0",
     "eslint-plugin-prettier": "^3.1.1",
     "jsdoc": "^3.6.3",
-    "libp2p-crypto": "^0.17.3",
-    "mocha": "^6.2.2",
+    "libp2p-crypto": "^0.17.5",
```
Comment (carsonfarmer) [inline package.json:87]: So we still need this as a top-level dependency, or is this left-over from when it was missing in threads-core?

File: src/ReadTransaction.ts [hunk 4]
```diff
@@ -100,7 +100,7 @@ export class ReadTransaction extends Transaction<ReadTransactionRequest, ReadTra
           resolve()
         } else {
           const ret: Instance<T> = {
-            instance: JSON.parse(reply.toObject().instance as string),
+            instance: JSON.parse(fromBase64(reply.toObject().instance as string)),
```
Comment (carsonfarmer) [inline src/ReadTransaction.ts:103]: We we end up using Buffer here elsewhere anyway, I think we could just stick with Buffer and avoid having those extra b64-lite deps?

File: src/WriteTransaction.ts [hunk 5]
```diff
@@ -174,7 +171,7 @@ export class WriteTransaction extends Transaction<WriteTransactionRequest, Write
           resolve()
         } else {
           const ret: Instance<T> = {
-            instance: JSON.parse(reply.toObject().instance as string),
+            instance: JSON.parse(fromBase64(reply.toObject().instance as string)),
```
Comment (carsonfarmer) [inline src/WriteTransaction.ts:174]: Same here, I think `Buffer.from(string, 'base64')` works

**Comment by andrewxhill** [comment]
> LGTM! There are just two comments that could probably use tickets? Your call.

took care for them, will be gtg once those tests pass

**Pull request merged**